### ISSUE TITLE
unable to show Image previews

### ISSUE THREAD

**Issue by sanwacompany** [teableio/teable#885]
**Describe the bug**
![微信截图_20240904134814](https://github.com/user-attachments/assets/5f25cd32-bae3-4fcd-8465-1d21e6204296)

**To Reproduce**
self-host (docker-compose), upload any image file, there is no preview.  it works on teable.io

environment " - NEXT_ENV_IMAGES_ALL_REMOTE=true" has been set in docker-compose.yml, as https://github.com/teableio/teable/issues/512  and https://github.com/teableio/teable/issues/429 guided.
I also tried to modify next.config.js images part like this and build my custom image:

images: {  
    deviceSizes: [640, 750, 828, 1080, 1200, 1920, 2048, 3840],  
    imageSizes: [16, 32, 48, 64, 96, 128, 256, 384],  
    minimumCacheTTL: 60,  
    formats: ['image/webp'],  
    loader: 'default',  
    dangerouslyAllowSVG: false,  
    disableStaticImages: false,  
    contentSecurityPolicy: "default-src 'self'; script-src 'none'; sandbox;",  
    remotePatterns: [  
      {  
        protocol: 'http',  
        hostname: '**',  
      },  
      {  
        protocol: 'https',  
        hostname: '**',   
      },  
    ],  
    unoptimized: false,  
  }, 
 

** Client (please complete the following information):**

- OS: windows 11, Docker-desktop 4.33.1
- Browser: Edge
- teable Version:Latest 

**Platform (Please tell us which deployment version you are using)**
docker-standalone

**Comment by tea-artist** [comment]
Can you provide the error screenshot in the browser console?

**Comment by sanwacompany** [comment]
![image](https://github.com/user-attachments/assets/0ca82d8f-2976-46fd-b302-ad2ff46ce167)
thanks for your help!  did u mean this?

**Comment by sanwacompany** [comment]
> Can you provide the error screenshot in the browser console?

![image](https://github.com/user-attachments/assets/07699b29-48ce-439a-ac46-b0781cd68087)
Thanks for your help, did u mean this?

**Comment by sanwacompany** [comment]
@tea-artist Hi boss, u still there?

**Comment by tea-artist** [comment]
It seems that the error is coming from your API.

<img width="1356" alt="image" src="https://github.com/user-attachments/assets/6ae7e58d-4251-4177-9398-9a7e8f83677c">

It shows your server(127.0.0.1:3003 looks not teable server) is blocking requests of not same origin, you should check our API server to solver this problem

**Comment by sanwacompany** [comment]
> It seems that the error is coming from your API.
> 
> <img alt="image" width="1356" src="https://private-user-images.githubusercontent.com/113977203/365086124-6ae7e58d-4251-4177-9398-9a7e8f83677c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2MTY4MDIsIm5iZiI6MTcyNTYxNjUwMiwicGF0aCI6Ii8xMTM5NzcyMDMvMzY1MDg2MTI0LTZhZTdlNThkLTQyNTEtNDE3Ny05Mzk4LTlhN2U4ZjgzNjc3Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwNlQwOTU1MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNThiZTIwNDI1YTRjYjNjYTZhNjZlZWYzZDA0MmNiZThjYzU2MzA3YWJkOTIxOTA1MjNmMjU3NThjMWRiODFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1Q8o60YD_8GzvVZCl-2giOF37tUAJ6lIEL28-vJHZGI">
> It shows your server(127.0.0.1:3003 looks not teable server) is blocking requests of not same origin, you should check our API server to solver this problem
![微信截图_20240906174532](https://github.com/user-attachments/assets/9258237f-9d68-4fd7-b70d-5338609c9d71)

@tea-artist thanks for your help. Initially, during deployment, I changed the port number to 3003 in the docker-compose file. Now, I have reverted to the default port 3000. This is the latest error screenshot, and NotSameOrigin seems to be the culprit. I have tried both the CE and EE versions, explored many parameters in the next.config.js file, used Nginx to proxy the service, and experimented with many Nginx configurations. I even used the Chrome CORS Unblocked plugin, but none of these have resolved the issue. It seems I haven't found the right direction to solve the problem.

Since I'm operating within a local network, I can use the lowest security settings. Should I continue modifying the contents in next.config.js, or what should I do?

**Comment by tea-artist** [comment]
Did you build the image by yourself after next.config.ts? if so you may need pass `ENABLE_CSP=false` as build args

**Comment by sanwacompany** [comment]
> Did you build the image by yourself after next.config.ts? if so you may need pass `ENABLE_CSP=false` as build args

I didn't rebuild. However, I mounted next.config.js onto the host machine, modified the content, then ran docker-compose down and docker-compose up -d. By using docker exec to check the next.config.js inside the container, I found that the content had been successfully modified.

I'll try to use helmet as stackoverflow said and try to build my custom image

**Comment by sanwacompany** [comment]
After 36 hours of attempts, I discovered it was actually quite simple: change ARG ENABLE_CSP=true to “false” in the Dockerfile. However, I also made the following changes: I added the helmet dependency in package.json, and in /apps/nestjs-backend/src/bootstrap.ts, I changed "app.use(helmet())" to 
`  app.use(helmet({
    crossOriginResourcePolicy: false,
  })` 
Then I built a custom Docker image, and the issue was resolved. I'm not sure if the latter two steps were unnecessary, but the problem was ultimately solved.

Thank you very much for contributing such a great project, and I'll support it to the best of my ability.
